hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/http/HttpServer2Metrics.java

```diff
@@ -91,4 +90,10 @@ public void getMetrics(MetricsCollector collector, boolean all) {
         .addGauge(HttpServer2MetricsInfo.HttpServerThreadQueueWaitingTaskCount,
             threadPool.getQueueSize());
   }
+
+  public void unRegister() {
```

Comment:
Should be `synchronized` to ensure consistent synchronization when accessing `instance`.  (I guess findbugs also complains about this.)